client/src/store/slices/catalogueSlice/catalogueSlice.js

```diff
@@ -1,52 +1,76 @@
 /* eslint-disable no-return-assign */
 /* eslint-disable no-param-reassign */
-import { createSlice } from '@reduxjs/toolkit';
-import axios from 'axios';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axiosConfig from '../../../axiosConfig';
+
+export const fetchCatalogue = createAsyncThunk('products/fetchCatalogue', async () => {});
+
+export const fetchCatalogueProducts = createAsyncThunk(
+  'products/fetchCatalogueProducts',
+  async (page, { rejectWithValue }) => {
+    try {
+      const { data, status } = await axiosConfig.get(`products/filter?perPage=5&startPage=${page}`);
+      if (status >= 200 && status <= 300) {
+        return data;
+      }
+    } catch (err) {
+      return rejectWithValue(err);
+    }
+  }
+);
+
+export const fetchPopularProducts = createAsyncThunk('products/isPopular', async (_, {rejectWithValue}) => {
+  try {
+    const { status, data } = await axiosConfig('/products/filter?isPopular=true');
+    if (status >= 200 && status <= 300) {
+      return data.products;
+    }
+  } catch (err) {
+    return rejectWithValue(err);
+  }
+});
 
 const catalogueSlice = createSlice({
   name: 'products',
   initialState: {
     products: [],
     popular: [],
     isLoading: false,
+    error: null,
+    totalPages: null,
+    currentPage: 1,
   },
   reducers: {
-    setProducts: (state, action) => {
-      state.products = action.payload;
-    },
-
     setPopular: (state, action) => {
       state.popular = action.payload;
     },
```

Comment:
i guess this reducer function is not needed anymore, please delete.

Reply:
solved